src/common.py

```diff
@@ -251,6 +251,68 @@ def v_space(n: int, col=None) -> None:
         else:
             st.write("#")
 
+def display_large_dataframe(df, 
+                            chunk_sizes: list[int] = [100, 1_000, 10_000]):
+    """
+    Displays a large DataFrame in chunks with pagination controls and row selection.
+
+    Args:
+        df: The DataFrame to display.
+        chunk_sizes: A list of chunk sizes to choose from.
+
+    Returns:
+        Selected rows from the current chunk.
+    """
+    # Dropdown for selecting chunk size
+    chunk_size = st.selectbox("Select Number of Rows to Display", chunk_sizes)
+    
+    # Calculate total number of chunks
+    total_chunks = (len(df) + chunk_size - 1) // chunk_size
+
+    # Initialize session state for pagination
+    if 'current_chunk' not in st.session_state:
+        st.session_state.current_chunk = 0
+
+    # Function to get the current chunk of the DataFrame
+    def get_current_chunk(df, chunk_size, chunk_index):
+        start = chunk_index * chunk_size
+        end = min(start + chunk_size, len(df))  # Ensure end does not exceed dataframe length
+        return df.iloc[start:end], start, end
+
+    # Display the current chunk
+    current_chunk_df, start_row, end_row = get_current_chunk(df, chunk_size, st.session_state.current_chunk)
+
+    event = st.dataframe(
+        current_chunk_df,
+        column_order=[
```

Comment:
This should be more flexible as the data displayed may vary. I think adding `column_order` as a function parameter would work best. Is there a reason why `selection_mode`, `on_select`, `use_container_width`, and `hide_index` have to be set to these specific values? If not I would suggest adding those as parameters as well.

Anything specific to a particular workflow/dataset should not be located in `common.py` as it is intended to contain functions of general use. As a sidenote, common modules should be moved in a separate directory to make this more clear to developers.

Reply:
That's a good point. These are specifically set because I came across this issue when using the pyopenms workflow for selecting spectra to plot when selecting a row, but this should be more flexible for other use cases.

We could pass `kwargs` to the `display_large_dataframe` method for `st.dataframe` because that's probably the only main method that would need to be changed based on particular needs. I opted for this option